src/App.js

```diff
@@ -1,23 +1,148 @@
-import React from 'react';
-import SentenceContainer from './components/Sentence/SentenceContainer.js'
-import SettingsContainer from './components/SettingsContainer.js'
-import logo from './logo.svg';
+import React, { Component } from 'react';
+import Sentence from './components/Sentence/Sentence'
+import Settings from './components/Settings'
 import './App.css';
+import {allScales, getAccidentalFromNoteString, isInArray, keyFormulas, nashvilleNumbers} from "./Constants";
 
-function App() {
-  return (
-    <div className="App">
-      <header className="App-header">
-          <SentenceContainer/>
-      </header>
-        <div>
-            TODO:
-            <li>- SHow guess value in setence with green or red correct/incorrect</li>
-                <li>- settings/config</li>
-        </div>
-        <SettingsContainer/>
-    </div>
-  );
+class App extends Component
+{
+
+    getAllKeysWithTrueValue = (obj) => {
+        let out = [];
+        let keys = Object.keys(obj);
+        for (let i of keys) {
+            if (obj[i] === true){
+                out.push(i);
+            }
+        }
+        return out;
+    }
+
+    componentWillMount() {
+        this.resetCurrentKeysByFilters();
+    }
+
+    resetCurrentKeysByFilters() {
+        let filterScalesByAccidental = (scaleContainer, allowedAccidentals = ['sharp', 'natural', 'flat']) => {
+            let filteredScales = [];
+            // TODO: Refactor this to a single if with all conditions in a single test
+            // TODO: Refactor this to use accidentals object instead of string matching
+            scaleContainer.forEach(function (scale) {
+                if ((scale[0].includes('#')) && isInArray('sharp', allowedAccidentals)) {
+                    filteredScales.push(scale);
+                }
+                else if ((scale[0].includes('b')) && isInArray('flat', allowedAccidentals)) {
+                    filteredScales.push(scale);
+                }
+                else if (((!scale[0].includes('#')) && (!scale[0].includes('b'))) && isInArray('natural', allowedAccidentals)) {
+                    filteredScales.push(scale);
+                }
+            });
+            return filteredScales;
+        }
+
+        let filteredMajorScales = filterScalesByAccidental(allScales.major, this.getAllKeysWithTrueValue(this.state.keyAccidentals));
+        let filteredMinorScales = filterScalesByAccidental(allScales.minor, this.getAllKeysWithTrueValue(this.state.keyAccidentals));
+
+        let allFilteredScales = {
+            major: filteredMajorScales,
+            minor: filteredMinorScales,
+        };
+
+        let buildFilteredKeys = (keyQuality) => {
```

Comment:
@yattias Would you mind taking a deep look at this function.  I feel like this can be rewritten without the nested fors, using map, or something.